### DISCUSSION TITLE
Use the pre-trained global model weights in Prostate Segmentation example

### DISCUSSION THREAD

**Discussion by Tarun-Khilani** [NVIDIA/NVFlare#1515] [Q&A] [answered]
Hi, I am trying to run the Prostate segmentation example - FedAvg Job but I would like to use a Pre-trained model. I have tried specifying the path in the "config_fed_server.json" by adding the "source_ckpt_file_full_name" in the args of the "PTFileModelPersistor" persistor and also tried to specify it using the "environments.json" file created inside the app folder in the job which contains -
{
    "APP_CKPT_DIR": "model",
    "APP_CKPT": "pretrained_model.pt"
}
It is still not using the pre-trained model and gives ExecutionException in the case of source_ckpt.

**Comment by chesterxgchen** [comment]
@ZiyueXu77  can you help when you are free?

**Comment by ZiyueXu77** [comment, marked as answer]
Hi @Tarun-Khilani , the "environment.json" is more for changing the default saving path of model persistence to a customized one. For starting from a pre-trained model, you can use "source_ckpt_file_full_name", add it to the persistor specs in config_fed_server.json. For example:

> {
>       "id": "persistor",
>       "name": "PTFileModelPersistor",
>       "args": {
>         "model": {
>           "path": "monai.networks.nets.segresnet.SegResNet",
>           "args": {
>             "blocks_down": [
>               1,
>               2,
>               2,
>               4
>             ],
>             "blocks_up": [
>               1,
>               1,
>               1
>             ],
>             "init_filters": 16,
>             "in_channels": 4,
>             "out_channels": 3,
>             "dropout_prob": 0.2
>           }
>         },
>         "source_ckpt_file_full_name": "/media/Experiment/NVFlare/workspace/init_model/init_global_model.pt"
>       }
>    }
